mainargs/src-3/Macros.scala

```diff
@@ -89,8 +89,11 @@ object Macros {
     val argSigs = Expr.ofList(argSigsExprs)
 
     val invokeRaw: Expr[(B, Seq[Any]) => T] = {
-      def callOf(args: Expr[Seq[Any]]) = call(method, '{ Seq( ${ args }) }).asExprOf[T]
-      '{ ((b: B, params: Seq[Any]) => ${ callOf('{ params }) }) }
+
+      def callOf(methodOwner: Expr[Any], args: Expr[Seq[Any]]) =
```

Comment:
`methodOwner` is the actual instance of an object that owns the main methods to call